lib/extractor/index.ts

```diff
@@ -1,13 +1,58 @@
-import { ImageType } from "../types";
+import {
+  getLayersFromPackages,
+  getPackagesFromRunInstructions,
+} from "../dockerfile/instruction-parser";
+import { AutoDetectedUserInstructions, ImageType } from "../types";
 import * as dockerExtractor from "./docker-archive";
 import * as ociExtractor from "./oci-archive";
 import {
+  DockerArchiveManifest,
   ExtractAction,
   ExtractedLayers,
+  ExtractedLayersAndManifest,
   ExtractionResult,
+  Extractor,
   FileContent,
+  ImageConfig,
+  OciArchiveManifest,
 } from "./types";
 
+export class InvalidArchiveError extends Error {}
```

Comment:
This is probably not important, but the convention is to add `this.name = "InvalidArchiveError"` in the constructor. I don't actually know why this is needed.